op/model.py

```diff
@@ -238,6 +242,14 @@ def _load(self):
         return self._backend.config_get()
 
 
+class Pod:
+    def __init__(self, backend):
+        self._backend = backend
+
+    def set_spec(self, spec, k8s_resources=None):
```

Comment:
I thought the idea was to not have something specific to k8s, but then...?

Reply:
I agree.  I think the issue was that there were some resources that some charms did need to manage but which were inherently K8s specific. So I think this was added as a way to clearly denote them as tied to a the specific underlying implementation.  I think there has been tension between the desire to model things in a more abstract way vs enabling the functionality that is needed to actually get things done considering the (TBH, really crappy) way that the K8s world works.  Specifically, a lot of images that we really want to use (e.g., kubeflow) are coded at a fundamental level to be very K8s aware and specific and depend on features that really have no analogue in the Juju model. But then the end result is stuff like this, which is ugly.